simpleflow/boto3_utils.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+import hashlib
+from contextvars import ContextVar
+from typing import Any
+
+import boto3
+
+from simpleflow.utils import json_dumps
+
+_client_var: ContextVar[dict | None] = ContextVar("boto3_clients", default=None)
```

Comment:
let's not put a default here
```suggestion
_client_var: ContextVar[dict] = ContextVar("boto3_clients")
```